src/teleport.cpp

```diff
@@ -23,34 +23,46 @@ static const flag_id json_flag_DIMENSIONAL_ANCHOR( "DIMENSIONAL_ANCHOR" );
 static const efftype_id effect_grabbed( "grabbed" );
 static const efftype_id effect_teleglow( "teleglow" );
 
+
 bool teleport::teleport( Creature &critter, int min_distance, int max_distance, bool safe,
                          bool add_teleglow )
 {
     if( min_distance > max_distance ) {
         debugmsg( "ERROR: Function teleport::teleport called with invalid arguments." );
         return false;
     }
-    Character *const p = critter.as_character();
-    const bool c_is_u = p != nullptr && p->is_avatar();
     int tries = 0;
     tripoint origin = critter.pos();
     tripoint new_pos = origin;
     map &here = get_map();
-    //The teleportee is dimensionally anchored so nothing happens
-    if( p && ( p->worn_with_flag( json_flag_DIMENSIONAL_ANCHOR ) ||
-               p->has_effect_with_flag( json_flag_DIMENSIONAL_ANCHOR ) ) ) {
-        p->add_msg_if_player( m_warning, _( "You feel a strange, inwards force." ) );
-        return false;
-    }
     do {
         int rangle = rng( 0, 360 );
         int rdistance = rng( min_distance, max_distance );
         new_pos.x = origin.x + rdistance * std::cos( rangle );
         new_pos.y = origin.y + rdistance * std::sin( rangle );
         tries++;
     } while( here.impassable( new_pos ) && tries < 20 );
+    teleport_to_point( critter, new_pos, safe, add_teleglow );
+}
```

Comment:
This function now doesn't return any value. Did you mean this?
```c++
return teleport_to_point( critter, new_pos, safe, add_teleglow );
```

Reply:
Yeah, not sure how I overlooked that.